assignments/array-methods.js

```diff
@@ -56,28 +56,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+fullName = runners.forEach((each) => {
+    return `${each.first_name} ${each.last_name}`
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+allCaps = runners.map((each) => {
+    return `${each.first_name.toUpperCase()}`
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts = runners.filter((each) => {
+    return each.shirt_size === "L"   
+})
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+ticketPriceTotal[0] = runners.reduce((start, current) => 
+  ({
+    "donation": start.donation + current.donation 
+  })
+)
+console.log(ticketPriceTotal)
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Find all runners whose first and last name have the same first character
+runners.filter((each) => (
+    each.first_name[0] === each.last_name[0]
+))
 
 // Problem 2
+// Calculate total donations coming from each company
+const getDonationsFromEachCompany = runners.sort((x, y) => {
+      if (x.company_name > y.company_name) return 1;
+      if (x.company_name < y.company_name) return -1;
+      return 0;
+    });
+    
+const res = [];
+    
+for (let i = 0; i < getDonationsFromEachCompany.length; i++) {
```

Comment:
For a complexity this high I must ask that you start developing a habit of using comments.